app/javascript/controllers/new_message_controller.js

```diff
@@ -0,0 +1,26 @@
+"use strict";
+
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["input", "submit"]
+
+  connect() {
+    this.inputTarget.focus()
+    this.inputTarget.addEventListener("keyup", this.disableSubmitButton.bind(this))
+    this.disableSubmitButton()
+  }
+
+  disconnect() {
+    this.inputTarget.removeEventListener("keyup", this.disableSubmitButton.bind(this))
+  }
+
+  disableSubmitButton() {
+    console.log(this.inputTarget.value.length)
+    if (this.inputTarget.value.length < 1) {
+      this.submitTarget.disabled = true
+    } else {
+      this.submitTarget.disabled = false
+    }
+  }
+}
```

Comment:
It seems a little surprising to have a stimulus controller solely for this form. My instinct would have been to either expand the form controller we already have to make it able to handle some of these extra features like initial focus and submit disable. But it's totally fine to leave this. We can either have separate form conrollers b/c maybe they're specific enough or we can generalize later after we have multiple forms.

Reply:
I hemmed and hawed about this. Stimulus seems like it'll be at it's best when controllers are tiny and single purposed, but it also seems to encourage reuse. The two conflict in my mind...

I opted for a dedicated controller here because this is the primary interaction widget on the screen and it's likely to be different from other forms we might have in many ways, and it's important to be robust.